diff --git a/node_modules/fast-deep-equal/react.js b/node_modules/fast-deep-equal/react.js
index 3489b98..ffe03c4 100644
--- a/node_modules/fast-deep-equal/react.js
+++ b/node_modules/fast-deep-equal/react.js
@@ -23,7 +23,6 @@ module.exports = function equal(a, b) {
 
     if (a.constructor === RegExp) return a.source === b.source && a.flags === b.flags;
     if (a.valueOf !== Object.prototype.valueOf) return a.valueOf() === b.valueOf();
-    if (a.toString !== Object.prototype.toString) return a.toString() === b.toString();
 
     keys = Object.keys(a);
     length = keys.length;
@@ -45,6 +44,8 @@ module.exports = function equal(a, b) {
       if (!equal(a[key], b[key])) return false;
     }
 
+    if (a.toString !== Object.prototype.toString) return a.toString() === b.toString();
+
     return true;
   }
 
